spec.html

```diff
@@ -38426,56 +38427,56 @@ <h1>Runtime Semantics: UnicodeEscape ( _C_ )</h1>
               1. Set _member_ to the string-concatenation of _member_ and _strP_.
               1. Append _member_ to _partial_.
           1. If _partial_ is empty, then
             1. Let _final_ be *"{}"*.
           1. Else,
-            1. If _gap_ is the empty String, then
+            1. If state_.[[Gap]] is the empty String, then
               1. Let _properties_ be the String value formed by concatenating all the element Strings of _partial_ with each adjacent pair of Strings separated with the code unit 0x002C (COMMA). A comma is not inserted either before the first String or after the last String.
               1. Let _final_ be the string-concatenation of *"{"*, _properties_, and *"}"*.
             1. Else,
-              1. Let _separator_ be the string-concatenation of the code unit 0x002C (COMMA), the code unit 0x000A (LINE FEED), and _indent_.
+              1. Let _separator_ be the string-concatenation of the code unit 0x002C (COMMA), the code unit 0x000A (LINE FEED), and _state_.[[Indent]].
               1. Let _properties_ be the String value formed by concatenating all the element Strings of _partial_ with each adjacent pair of Strings separated with _separator_. The _separator_ String is not inserted either before the first String or after the last String.
-              1. Let _final_ be the string-concatenation of *"{"*, the code unit 0x000A (LINE FEED), _indent_, _properties_, the code unit 0x000A (LINE FEED), _stepback_, and *"}"*.
-          1. Remove the last element of _stack_.
-          1. Set _indent_ to _stepback_.
+              1. Let _final_ be the string-concatenation of *"{"*, the code unit 0x000A (LINE FEED), _state_.[[Indent]], _properties_, the code unit 0x000A (LINE FEED), _stepback_, and *"}"*.
+          1. Remove the last element of _state_.[[Stack]].
+          1. Set _state_.[[Indent]] to _stepback_.
           1. Return _final_.
         </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-serializejsonarray" aoid="SerializeJSONArray">
-        <h1>Runtime Semantics: SerializeJSONArray ( _value_ )</h1>
-        <p>The abstract operation SerializeJSONArray with argument _value_ serializes an array. It has access to the _stack_, _indent_, and _gap_ values of the current invocation of the `stringify` method.</p>
+        <h1>Runtime Semantics: SerializeJSONArray ( _state_, _value_ )</h1>
+        <p>The abstract operation SerializeJSONArray with arguments _state_ and _value_ serializes an array. It performs the following steps:</p>
         <emu-alg>
-          1. If _stack_ contains _value_, throw a *TypeError* exception because the structure is cyclical.
-          1. Append _value_ to _stack_.
-          1. Let _stepback_ be _indent_.
-          1. Set _indent_ to the string-concatenation of _indent_ and _gap_.
+          1. If _state_.[[Stack]] contains _value_, throw a *TypeError* exception because the structure is cyclical.
```

Comment:
and?
```suggestion
          1. Assert: _state_ is a Record.
          1. If _state_.[[Stack]] contains _value_, throw a *TypeError* exception because the structure is cyclical.
```